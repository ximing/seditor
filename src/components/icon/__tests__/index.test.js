/**
 * Created by liz
 * Email: lizhengnacl@163.com
 * Tel: 18686768624
 * Date: 16/12/16
 */
'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Icon from '..';

describe('common-component', () => {
    it('Icon Snapshot Testing', () => {
        const component = renderer.create(
            <Icon type="wode" spin={true} className="test" style={{}}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Icon DOM Testing', () => {
        // Render a checkbox with label in the document
        const component = shallow(
            <Icon type="wode" spin={true} className="test" style={{}}/>
        );

        expect(component.text()).toEqual('');
        // component.find('input').simulate('change');
        // expect(component.text()).toEqual('On');
    });
});
