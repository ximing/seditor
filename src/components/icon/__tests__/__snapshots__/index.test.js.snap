exports[`common-component Icon Snapshot Testing 1`] = `
<i
  className="xm-icon iconfont icon-wode iconfont-spin test"
  onClick={[Function]}
  style={Object {}} />
`;
